iceoryx_utils/include/iceoryx_utils/posix_wrapper/pthread.hpp

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
+//
```

Comment:
Since you already created a file wouldn't it be better to go all the way and create a full posix thread wrapper. You do not have to go all the way with all the fancy features but a constructor which creates it then a destructors which cleans it up and some methods like `setName`, `getName` which then do not need the thread handle anymore.

All the files are already there and the code you have to add should be not too much and would fit perfectly into this cleanup story. Otherwise we have a cleanup story where we introduce again new technical debt.

Reply:
If we are sure, that we will replace all `std::thread` occurrences  with our own implementation, I'd agree to do that now. TBH I don't see this coming :-) `std::thread` is just fine and there are various safety certified versions from vendors available AFAIK. I'll leave this as it is.